src/status_im/ethereum/macros.clj

```diff
@@ -24,19 +24,10 @@
 
 (defn network->icon [network]
   (let [s     (str "./resources/images/tokens/" (name network) "/0-native.png")
-        s-js  (str "." s)
-        image (gensym)]
+        s-js  (str "." s)]
     (if (.exists (io/file s))
-      `(let [~image (atom nil)]
-         (fn []
-           (or @~image
-               (reset! ~image (js/require ~s-js)))))
-      `(let [~image (atom nil)]
-         (fn []
-           (or
-            @~image
-            (reset! ~image
-                    (js/require "../resources/images/tokens/default-token.png"))))))))
+      `(js/require ~s-js)
```

Comment:
could you elaborate on this change? im not sure i understand

Reply:
hi @flexsurfer,
Thank you very much for reviewing the PR.

`(network->icon network)` when icon exists will returns
```
(let [atom_name (atom nil)]
         (fn []
           (or @atom_name
               (reset! atom_name (js/require icon_path)))))
```

which is function definition not function call and this function will be assigned as :icon value

And final output of 
`(resolve-native-currency-icons {:mainnet {:symbol :eth}})`
will become
`=> {:mainnet {:symbol :eth, :icon {:source #function[user/eval4837/fn--4839]}}}`

Reply:
just wondering how it worked before then ?

Reply:
> NOTE: on 1.19 it is also an issue, so it is not regression

It didn't

Reply:
Or even before that? I am not sure :D

Reply:
yes, even before :)